hk_install_centos7.sh

```diff
@@ -0,0 +1,289 @@
+KillSignal=SIGTERM
+TimeoutStopSec=60
+ExecStart=/usr/bin/java -$xmx -Djava.net.preferIPv4Stack=true -jar $iri_v -c iota.ini
+SyslogIdentifier=IRI
+Restart=on-failure
+RestartSec=30
+
+[Install]
+WantedBy=multi-user.target
+Alias=iota.service
+EOF
+
+# Configure IRI
+cat <<EOF | su - iota -c "tee /home/iota/node/iota.ini"
+[IRI]
+PORT = 14265
+UDP_RECEIVER_PORT = 14600
+TCP_RECEIVER_PORT = 15600
+API_HOST = 0.0.0.0
+IXI_DIR = ixi
+HEADLESS = true
+DEBUG = false
+TESTNET = false
+DB_PATH = mainnetdb
+RESCAN_DB = false
+REMOTE_LIMIT_API = "interruptAttachingToTangle, attachToTangle, setApiRateLimit"
+# We don't need to add normal neighbors as we're going to be using Nelson
+EOF
+
+# Download the last known Tangle database
+clear
+echo ""
+echo "Downloading latest Tangle database"
+echo "Please be patient. This might take a while depending on your internet connection speed"
+echo ""
+echo ""
+curl -L http://db.iota.partners/IOTA.partners-mainnetdb.tar.gz -o /tmp/IOTA.partners-mainnetdb.tar.gz
+su - iota -c "tar xzfv /tmp/IOTA.partners-mainnetdb.tar.gz -C /home/iota/node/mainnetdb"
+rm /tmp/IOTA.partners-mainnetdb.tar.gz
+
+# Install Nelson
+npm install -g nelson.cli
+
+# Start IOTA and enable it on boot
+service iota start
+systemctl start iota.service
+systemctl enable iota.service
+
+# Install cronjob to automatically update IRI
+echo '*/15 * * * * "curl -s https://gist.githubusercontent.com/zoran/48482038deda9ce5898c00f78d42f801/raw | bash"' | tee /etc/cron.d/iri_updater > /dev/null
+
+# Start Nelson with pm2
+npm install pm2 -g
+pm2 startup
+pm2 start nelson -- --getNeighbors
+pm2 save
+
+
+#
+# IOTA/NELSON section end
+#
+
+#
+# Oyster/Hooknode section start
+#
+
+service iota stop
+
+# Install the latest 1.9.3 go
+# Hooknode won't compile if you're using v1.9.4 or newer
+rpm --import https://mirror.go-repo.io/centos/RPM-GPG-KEY-GO-REPO
+curl -s https://mirror.go-repo.io/centos/go-repo.repo | tee /etc/yum.repos.d/go-repo.repo
+
+# Comment this out if newer than 1.9.3 becomes supported, see next few lines for your options
+yum install -y golang-1.9.3-0.el7.centos
+yum versionlock golang-*
+
+# Uncomment these two lines if newer than 1.9.3 becomes supported but not 1.10.x
+#goVersionNumber=`yum --showduplicates list golang |grep -oP '1.9\.[0-9].*centos?' | sort |tail -n 1`
+#yum install -y golang-$goVersionNumber
+
+# Uncomment this line if 1.10.x ever becomes supported
+#yum install -y golang
+
+# Create a user for hooknode and setup go
+adduser hooknode
+
+su - hooknode -c "echo \"export GOROOT=/usr/lib/golang\" >> /home/hooknode/.bash_profile"
+su - hooknode -c "echo \"export GOPATH=/home/hooknode/go\" >> /home/hooknode/.bash_profile"
+su - hooknode -c "echo \"export PATH=/home/hooknode/go/bin:\$PATH\" >> /home/hooknode/.bash_profile"
+
+su - hooknode -c "mkdir -p /home/hooknode/go/src"
+su - hooknode -c "mkdir -p /home/hooknode/go/bin"
+
+# Install the go dependancy manager
+su - hooknode -c "go get -u github.com/golang/dep/cmd/dep"
```

Comment:
dep is already installed with the rest of the dependencies when you call `make install-deps`. It certainly doesn't hurt to have it before that, but I think it's superfluous.

Reply:
Dep is not automatically installed on CentOS when you install golang. I had to manually install it to get things working otherwise `make install-deps` would fail.